tests/frontend/test-class-wpseo-opengraph-image.php

```diff
@@ -384,11 +385,128 @@ public function test_get_images_from_content() {
 		$this->go_to( get_permalink( $post_id ) );
 		$class_instance = $this->setup_class();
 
+		// We only expect our attachment image to appear in the results.
 		$expected = $this->sample_full_file_array( $attached_image, $attach_id );
-		$expected['https://cdn.yoast.com/app/uploads/2018/03/Caroline_Blog_SEO_FI-600x314.jpg'] = array(
-			'url' => 'https://cdn.yoast.com/app/uploads/2018/03/Caroline_Blog_SEO_FI-600x314.jpg',
+
+		$this->assertEquals( $expected, $class_instance->get_images() );
+	}
+
+	/**
+	 * Test getting the image from post content.
+	 */
+	public function test_get_images_from_content_cdn_first() {
+
+		// Create our post.
+		$post_id = $this->create_post();
+
+		// Upload an image to it.
+		$image          = '/assets/yoast.png';
+		$upload_dir     = wp_upload_dir();
+		$basename       = basename( $image );
+		$source_image   = dirname( __FILE__ ) . '/..' . $image;
+		$featured_image = $upload_dir['path'] . '/' . $basename;
+		copy( $source_image, $featured_image ); // Prevent original from deletion.
+
+		$file_array = array(
+			'name'     => $basename,
+			'tmp_name' => $featured_image,
+		);
+		$attach_id  = media_handle_sideload( $file_array, $post_id );
+
+		// Get the image URL so we can add it in the post content.
+		$file           = get_attached_file( $attach_id );
+		$attached_image = $upload_dir['url'] . '/' . basename( $file );
+		$image2_url     = 'https://cdn.yoast.com/app/uploads/2018/03/Caroline_Blog_SEO_FI-600x314.jpg';
+
+		// Update the post content.
+		$post_content = '<p>This is a post. It has an image hosted on a cdn:</p>
+<img src="' . $image2_url . '"/>
+<p>It also has an image that is attached to this post:</p>
+<img src="' . $attached_image . '"/>
+<p>End of post</p>';
+
+		wp_update_post(
+			array(
+				'ID'           => $post_id,
+				'post_content' => $post_content,
+			)
+		);
+
+		// Run our test.
+		$this->go_to( get_permalink( $post_id ) );
+		$class_instance = $this->setup_class();
+
+		// We only expect our attachment image to appear in the results.
+		$expected = $this->sample_full_file_array( $attached_image, $attach_id );
+
+		$this->assertEquals( $expected, $class_instance->get_images() );
+	}
+
+	/**
+	 * Test if only one image is returned when multiple are available.
+	 *
+	 * @covers WPSEO_OpenGraph_Image::add_content_images()
+	 */
+	public function test_get_only_one_image_from_content() {
+
+		// Create our post.
+		$post_id = $this->create_post();
+
+		// Upload an image to it.
```

Comment:
These pieces of logic have a lot of similarities, they should be converted into a method which will handle the file attachments.
Also the code in `test_get_images_from_content_cdn_first`